plutus-core/testlib/Test/Tasty/Extras.hs

```diff
@@ -21,32 +24,47 @@ import Data.ByteString.Lazy qualified as BSL
 import Data.Text (Text)
 import Data.Text.Encoding (encodeUtf8)
 import Data.Version
-import System.FilePath ((</>))
+import System.FilePath (joinPath, (</>))
 import System.Info
 import Test.Tasty
 import Test.Tasty.Golden
 
+ghcVersion :: String
+ghcVersion = showVersion compilerVersion
+
+-- Given two lists of FilePaths, concatenate the members of the first list
+-- together, append the GHC version number, then append the concatenation of the
+-- members of the second list.
+makeVersionedFilePath :: [FilePath] -> [FilePath] -> FilePath
+makeVersionedFilePath left right = joinPath left </> ghcVersion </> joinPath right
+
 -- | A 'TestTree' of tests under some name prefix.
-type TestNested = Reader [String] TestTree
+type TestNested = Reader [FilePath] TestTree
 
 -- | Run a 'TestTree' of tests with a given name prefix.
-runTestNestedIn :: [String] -> TestNested -> TestTree
```

Comment:
I''m not sure if I like this name.  It sounds like it runs a test, but really it's running a computation in the Reader monad.